04_library_management_system.rb

```diff
@@ -0,0 +1,125 @@
+class Member
+  attr_accessor :name, :book_issued, :status
+
+  def initialize(name)
+    @name = name
+    @status = "No book taken"
+  end
+end
+
+class Book
+  attr_accessor :name, :issued_to, :status
+
+  def initialize(name)
+    @name = name
+    @status = "New Book"
+  end
+
+  def issue_to_member(member)
+    if (@issued_to.nil?() || @status == "Available")
+      if (member.status == "No book taken")
```

Comment:
@rohitjoshixyz Implement methods to check status. `member.book_taken?` or `member.book_available?` This makes our code future-proof as if you want to change status string in future we have to change it only at one place only.